### DISCUSSION TITLE
Navigation inside the same page.

### DISCUSSION THREAD

**Discussion by lazaroofarrill** [quasarframework/quasar#9938] [General - Components / Directives / etc] [answered]
I can't scroll to an element in the same page using the id property. Is there a specific way to achieve this in Vue?

**Comment by justinbeaty** [comment]
How are you trying to scroll? You should be able to just do:

```
<a href="#sometarget">Go to some target</a>
```

But if you're trying to do something else like do it without a user click you'll have to tell us, we can't read minds ;).

Maybe also see the [Scrolling Utils](https://quasar.dev/quasar-utils/scrolling-utils) part of the docs.

**Reply by lazaroofarrill** [reply]
I already tried a regular link, it doesn't work. I think it has something to do with the router. I did manage to achieve what I wanted with scrolling utils though, thanks a lot.

**Comment by lazaroofarrill** [comment, marked as answer]
@justinbeaty Finally navigation with regular anchors is working. I had to replace default scroll behavior with this and set router to history mode.

```
scrollBehavior: (to, from, savedPosition) => {
      if (to.hash) {
        return {
          el: to.hash,
          behavior: 'smooth',
          top: 65
        };
      }

```

**Reply by hawkeye64** [reply]
I typically set the router `scrollBehavior` to this (Vue 3, but you can extrapolate it back to vue 2):

```js
    scrollBehavior (to, from, savedPosition) {
      return new Promise(resolve => {
        setTimeout(() => {
          if (to.hash !== undefined && to.hash !== '' && document) {
            const el = document.getElementById(to.hash.substring(1))

            if (el !== null) {
              resolve({ left: 0, top: el.offsetTop - el.scrollHeight })
              return
            }
          }

          resolve(savedPosition || { left: 0, top: 0 })
        }, 100)
      })
    },
```
